### PR TITLE
spm cl error

### PR DESCRIPTION
**Pull request by MasterSwift**
Error probably from package name in package.swift not being the same as the repo.

error: invalid target name at 'PromiseTests'; name of non-test targets cannot end in 'Tests'

Added the test target.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import PackageDescription
 
 let package = Package(
-    name: "Promises",
+    name: "Promise",
```
Comment (khanlou) [inline Package.swift:5]: what effect does changing this package name have? i was trying to make sure the package name was the same as the cocoapods name.

**Comment by isaac-weisberg** [comment]
Checked out, this doesn't even fix the issue, but thanks for the tip, tests are really not declared properly for SPM environment.

File: Package.swift (hunk 1 above)
Comment (isaac-weisberg) [reply Package.swift:5]: Changes to package name in Swift Package Manager don't change anything.

Comment (MasterSwift) [reply Package.swift:5]: yea it didnt fix the issue. wasn't sure why but It seems you debunked the linux testing issue! 👍

**Comment by isaac-weisberg** [comment]
This pull request's contents are featured in #38 with massive improvements

**Pull request merged**